src/dialects/postgres/query.js

```diff
@@ -51,6 +51,13 @@ export class PostgresQuery extends AbstractQuery {
         || /Unable to set non-blocking to true/i.test(error)
         || /SSL SYSCALL error: EOF detected/i.test(error)
         || /Local: Authentication failure/i.test(error)
+
+        // If the query fails with a client-side timeout, invalidate the connection because
+        // it's probably stuck. (Note that query_timeout should be greater than
```

Comment:
You mention that `query_timeout` should be greater than `statement_timeout` but what happens if it's not? Is this something we should document or is this already a documented best practice by https://github.com/brianc/node-postgres ?

Reply:
The purpose of `query_timeout`, I think, is to help the client know when the connection is stuck because it sometimes takes quite a long time for it to realize that on its own (15 minutes in our environment!). We know it's a stuck connection if the query takes longer than the statement timeout, because otherwise we would have seen a statement timeout.

So setting `query_timeout` < `statement_timeout` means that we'll never see statement timeouts and we'll invalidate connections unnecessarily.

This should probably be documented in node-postgres but isn't. (Or maybe it's considered obvious enough?)